internal/pkg/mapper/mapper.go

```diff
@@ -1,33 +1,40 @@
 package mapper
 
 import (
+	"github.com/keboola/keboola-as-code/internal/pkg/filesystem"
+	"github.com/keboola/keboola-as-code/internal/pkg/filesystem/fileloader"
 	"github.com/keboola/keboola-as-code/internal/pkg/model"
 	"github.com/keboola/keboola-as-code/internal/pkg/utils"
 )
 
-// LocalSaveMapper to modify how the object will be saved in the filesystem.
+// LocalSaveMapper is intended to modify how the object will be saved in the filesystem.
 type LocalSaveMapper interface {
 	MapBeforeLocalSave(recipe *model.LocalSaveRecipe) error
 }
 
-// LocalLoadMapper to modify/normalize the object internal representation after loading from the filesystem.
+// LocalLoadMapper is intended to modify/normalize the object internal representation after loading from the filesystem.
 // Note: do not rely on other objects, they may not be loaded yet, see OnLocalChangeListener.
 type LocalLoadMapper interface {
 	MapAfterLocalLoad(recipe *model.LocalLoadRecipe) error
 }
 
-// RemoteSaveMapper to modify how the object will be saved in the Storage API.
+// RemoteSaveMapper is intended to modify how the object will be saved in the Storage API.
 type RemoteSaveMapper interface {
 	MapBeforeRemoteSave(recipe *model.RemoteSaveRecipe) error
 }
 
-// RemoteLoadMapper to modify/normalize the object internal representation after loading from the Storage API.
+// LocalFileLoadMapper is intended to modify file load process.
+type LocalFileLoadMapper interface {
+	LoadLocalFile(def *filesystem.FileDef, fileType filesystem.FileType, next filesystem.LoadHandler) (filesystem.File, error)
+}
+
+// RemoteLoadMapper is intended to modify/normalize the object internal representation after loading from the Storage API.
```

Comment:
Mapper moze implementovat metodu `LoadLocalFile`, ktorou sa da modifikovat, ako bude subor nacitany.
Metoda obsahuje parameter `next` , ktory je odkaz na nasledujuci mapper.
Posledny mapper dostane ako `next` `defaultHandler`.

Je to inspirovane vzorom middleware.
Takto je to flexibilne, ... callback `next` mozem/nemusim pouzit.
Po zavolani `next` mozem hodnotu este upravit.

Vid testy dalej.